File: src/main/java/com/ggardet/doctor/internal/repository/DoctorRepository.java
package com.ggardet.doctor.internal.repository;

import com.ggardet.doctor.internal.entity.Doctor;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface DoctorRepository extends JpaRepository<Doctor, UUID> {
    List<Doctor> findAllByHospitalId(final UUID hospitalId);

    List<Doctor> findAllBySpecialty(final String specialty);
}

File: src/main/kotlin/com/ggardet/doctor/internal/repository/DoctorRepository.kt
package com.ggardet.doctor.internal.repository

import com.ggardet.doctor.internal.entity.Doctor
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface DoctorRepository : JpaRepository<Doctor, UUID> {
    fun findAllByHospitalId(hospitalId: UUID): List<Doctor>
    
    fun findAllBySpecialty(specialty: String): List<Doctor>
}